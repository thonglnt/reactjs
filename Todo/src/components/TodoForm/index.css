.add-button {
  padding: 2px 8px;
}

.add-todo {
  display: flex;
}

.add-icon {
  display: flex;
  width: 30px;
  height: 30px;
}

.todo-title {
  font-size: 70px;
  color: var(--chinese-silver);
  text-transform: lowercase;
}

@media screen and (max-width: 992px) {
  .add-icon {
    width: 25px;
    height: 25px;
  }
}

@media screen and (max-width: 775px) {
  .add-icon {
    width: 20px;
    height: 20px;
  }

  .todo-title {
    font-size: 40px;
  }
}
