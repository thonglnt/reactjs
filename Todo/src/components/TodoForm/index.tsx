// Lib
import { FormEvent, useContext } from 'react';

// Assets
import { iconAdd } from '@assets';

// Component
import Input from '@components/Input';
import Spinner from '@components/Spinner';

// Contexts
import { TodoContext } from '@contexts/todoContext';

export type TodoFormProps = {
  onAddTodo: (event: FormEvent<HTMLFormElement>) => void;
};

const TodoForm = ({ onAddTodo }: TodoFormProps) => {
  const { isAddingTodo } = useContext(TodoContext);

  return (
    <>
      <h1 className='title todo-title'>todos</h1>
      <form data-testid='form-todo' method='post' onSubmit={onAddTodo}>
        <div className='add-todo'>
          <Input
            size='input-lg'
            id='task'
            name='task'
            placeholder='Add new todo'
            defaultValue=''
            label=''
            isRequired={true}
          />
          <button className='add-button'>
            {isAddingTodo ? (
              <Spinner className='loading-add-todo' />
            ) : (
              <img className='add-icon' src={iconAdd} alt='add-todo' />
            )}
          </button>
        </div>
      </form>
    </>
  );
};

export default TodoForm;
