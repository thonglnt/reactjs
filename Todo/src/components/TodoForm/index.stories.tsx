// Lib
import { Story } from '@storybook/react';
import { action } from '@storybook/addon-actions';

// Components
import TodoForm, { TodoFormProps } from '@components/TodoForm';
import BodyContent from '@components/BodyContent';

export default {
  title: 'TodoForm',
  component: TodoForm,
};

export const TodoFormSample: Story<TodoFormProps> = () => (
  <BodyContent>
    <TodoForm onAddTodo={action('onAddTodoItem')} />
  </BodyContent>
);
