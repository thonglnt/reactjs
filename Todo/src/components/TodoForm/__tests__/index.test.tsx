// Lib
import { render, screen, fireEvent } from '@testing-library/react';

// Components
import TodoForm, { TodoFormProps } from '@components/TodoForm';

describe('Test TodoForm component', () => {
  const initTodoForm: TodoFormProps = {
    onAddTodo: jest.fn(),
  };

  test('It should render correctly with initial Props', () => {
    const { container } = render(<TodoForm {...initTodoForm} />);

    expect(container).toMatchSnapshot();
  });

  test('It should call remove todo', () => {
    render(<TodoForm {...initTodoForm} />);
    fireEvent.submit(screen.getByTestId('form-todo'));

    expect(initTodoForm.onAddTodo).toHaveBeenCalledTimes(1);
  });
});
