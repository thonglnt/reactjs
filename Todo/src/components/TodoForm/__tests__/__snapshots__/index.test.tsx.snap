// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TodoForm component It should render correctly with initial Props 1`] = `
<div>
  <h1
    class="title todo-title"
  >
    todos
  </h1>
  <form
    data-testid="form-todo"
    method="post"
  >
    <div
      class="add-todo"
    >
      <label
        class="input-label"
        for="task"
      />
      <div
        class="input-wrapper"
      >
        <input
          class="input input-lg"
          id="task"
          name="task"
          placeholder="Add new todo"
          required=""
          type="text"
          value=""
        />
      </div>
      <button
        class="add-button"
      >
        <img
          alt="add-todo"
          class="add-icon"
          src="IMAGE_MOCK"
        />
      </button>
    </div>
  </form>
</div>
`;
